app/utils/timezone.js

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import DeviceInfo from 'react-native-device-info';
+import {updateUser} from 'app/actions/views/edit_profile';
+import {mattermostStore} from 'app/mattermost';
+
+export function getSupportedTimezones(supportedTimezones) {
+    if (supportedTimezones) {
+        return supportedTimezones.split(',');
+    }
+    return [];
+}
+
+export function getDeviceTimezone() {
+    return DeviceInfo.getTimezone();
+}
+
+function updateUserTimezone(user) {
+    const {dispatch, getState} = mattermostStore;
+    updateUser(user)(dispatch, getState);
```

Comment:
see comment above but...

when doing this we should be changing them to `dispatch(action(args))` that way the action can be a regular action or a thunk and we don't really need to know